symbolic-symcache/src/new/raw.rs

```diff
@@ -67,6 +68,19 @@ pub struct Function {
     pub lang: u32,
 }
 
+impl PartialEq for Function {
+    fn eq(&self, other: &Self) -> bool {
+        self.name_offset == other.name_offset && self.comp_dir_offset == other.comp_dir_offset
```

Comment:
`lang` seems like something that might be important to include here, in case a function somehow has some parts in say, C# and other sections in C++ (looking at you, usyms). I excluded it for now just because I'm guessing the expectation is that all functions in a single symcache are usually for one common language, but I could be wrong.